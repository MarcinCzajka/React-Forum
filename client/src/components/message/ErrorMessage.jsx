
import React from 'react';
import { Message } from 'semantic-ui-react';

export const ErrorMessage = props => {
    console.log(props)
    if(props.message) {
        return (
            <Message 
                error
                header = 'Error'
                content = {props.message} 
            />
        ) 
    } else {
        return ''
    }
}