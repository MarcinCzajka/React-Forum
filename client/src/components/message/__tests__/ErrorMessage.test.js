import React from 'react';
import { create } from "react-test-renderer";
import ErrorMessage from '../ErrorMessage';

describe('ErrorMessage component', () => {
   test('match snapshot with error message provided in props', () => {
       const component = create(<ErrorMessage message='This is serious error message' />);
       expect(component).toMatchSnapshot();
   });

   test('return empty string if no error message is provided', () => {
      const component = create(<ErrorMessage />);
      expect(component).toMatchSnapshot();
  });

});