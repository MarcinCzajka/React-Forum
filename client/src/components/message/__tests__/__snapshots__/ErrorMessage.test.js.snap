// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component match snapshot with error message provided in props 1`] = `
<div
  className="ui error message"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Error
    </div>
    <p>
      This is serious error message
    </p>
  </div>
</div>
`;

exports[`ErrorMessage component return empty string if no error message is provided 1`] = `""`;
